go/vt/sqlparser/analyzer.go

```diff
@@ -146,6 +146,8 @@ func Preview(sql string) StatementType {
 		return StmtUpdate
 	case "delete":
 		return StmtDelete
+	case "savepoint":
+		return StmtSavepoint
 	}
```

Comment:
there is savepoint present in the same method with another switch case. Either remove this or remove that.